lib/itertools/stream/stream.go

```diff
@@ -404,3 +446,52 @@ func MergeStreams[T any](
 		}
 	}
 }
+
+// Until iterates the stream, returning early if a predicate return true for a given item.
+func Until[T any](stream Stream[T], predicate func(T) bool) Stream[T] {
```

Comment:
nit: consider renaming to `TakeWhile` and having `true` be the continue signal and `false` be the halt signal.  I believe that is more conventional, at least thats how I've always seen combinators of this kind done.